Allow for game-type choice

ksabbak opened the pull request:
Game can now be played human vs. human or human vs. computer.

ksabbak committed:
Allow for game-type choice

Game can now be played human vs. human or human vs. computer.

hnlee commented on src/clojure_tictactoe/cli/output/instructions_printer.clj:
```diff
@@ -21,7 +21,18 @@ Examples:
 (defn print-marker-instructions [player]
   (println (str "What marker would you like for " player "?")))
+
+(defn stringify-options [options]
```
This name is a bit misleading since you're doing more than just converting the list to a string.

ksabbak replied:
Good point.

hnlee commented on test/clojure_tictactoe/cli/input/input_getter_test.clj:
```diff
@@ -12,18 +13,29 @@
+    (deftest get-game-type-test-display
+      (testing "Prints proper message"
+        (is (string/includes? (with-out-str (with-in-str "1" (get-game-type ["Pie" "Cake" "Ice cream"]))) instructions-printer/game-choice-message))
+        (is (string/includes? (with-out-str (with-in-str "1" (get-game-type ["Pie" "Cake" "Ice cream"]))) "1. Pie\n2. Cake\n3. Ice cream"))))
```
Not sure what the distinction is between these two assertions.

ksabbak replied:
Just different parts of the proper message.

hnlee commented on src/clojure_tictactoe/game/game_play.clj:
```diff
@@ -9,33 +9,34 @@
-        markers (players/acquire-both-markers '("you" "the computer"))]
+  (let [game-type (input-getter/get-game-type game-options)
+        board (board/render-empty-board standard-board-size)
+        markers (players/acquire-both-markers '("player 1" "player 2"))]
```
My comment on your last PR about hard-coded values was actually in reference to this list.

ksabbak replied:
Oooh! Yeah and that is a bit of a problem in this version isn't it. >_> I'll fix it.

hnlee commented on src/clojure_tictactoe/cli/input/input_getter.clj:
```diff
@@ -27,17 +27,26 @@
-            (recur))))))
+    (println instructions-printer/game-choice-request)
+    (let [choice-limit (+ (count options) 1)]
+      (if-let [choice (parse-move-input (get-user-input) choice-limit)]
```
I think you need to rename `parse-move-input` if you're going to use it to parse game type input as well as move input.

I'm wondering if you could just pass the list of choices (either the game types or the board) directly to this input parsing function and have it calculate what the limit should be instead of doing the `(+ (count options) 1)` in this function. I'm also wondering if that same input parsing function can subtract 1 before returning so you don't have to do that in the next line either, though maybe not if your board starts with 0.

ksabbak replied:
Yeah, I was thinking about that. I think that's something I might change in the future since it makes sense for the functions and I think as far as ux goes, starting with 1 instead of 0 is probably better. Thanks!

ksabbak committed:
Fix function name, hard-coding

In response to comments on  https://github.com/ksabbak/ClojureTTT/pull/16

Changes to players 'hardcoding' is more of a band-aid than a permanent fix
